src/message.js

```diff
@@ -104,43 +104,44 @@ function replaceElements(props, formatted) {
             [].splice.apply(sourceArray, [start, deleteCount].concat(itemsArray));
         }
 
-        return Object.keys(elements).reduce((ret, key) => {
+        return Object.keys(elements).reduce((nodes, key) => {
             const element = elements[key];
 
-            ret.forEach((str, i) => {
-                // Insert array into the correct ret position.
-                function replaceRetItem(array) {
-                    splice(ret, i, 1, array);
-                }
+            // Insert array into the correct ret position.
+            function replaceNode(array, i) {
+                splice(nodes, i, 1, array);
+            }
 
-                if (typeof str !== "string") {
-                    return; // continue;
+            for (let i = 0; i < nodes.length; i += 1) {
```

Comment:
Great catch! Can we include a comment here with a short summary saying why we can't use forEach? Something like: `// Note can't use .forEach() because array length might change during loop.` or something better :)

Reply:
ok. You'll notice I got rid of this ugly imperative coding in the next commit. Way safer.

Reply:
Ignore my message asking to include a comment... This code has been modified anyway in subsequent commits.